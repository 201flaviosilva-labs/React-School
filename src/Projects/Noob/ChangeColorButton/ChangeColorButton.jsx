import React, { useState } from "react";
import NoobProject from "../../../Pages/Noob/components/Project";
import { InputLabel, Checkbox } from "@mui/material";

export default function ChangeColorButton() {
	const [color, setColor] = useState("#ff0000");
	const text = color === "#ff0000" ? "Red" : "Green";

	const handleClick = () => {
		setColor(color === "#ff0000" ? "#00ff00" : "#ff0000");
	};

	return (
		<NoobProject title="Change Color Button" className="NoobChangeColorButton">
			<button
				onClick={handleClick}
				style={{ backgroundColor: color, }}>
				{text}
			</button>

			<InputLabel>
				<Checkbox
					checked={color !== "#ff0000"}
					onChange={handleClick}
				/> {text}
			</InputLabel>


		</NoobProject>
	)
}
