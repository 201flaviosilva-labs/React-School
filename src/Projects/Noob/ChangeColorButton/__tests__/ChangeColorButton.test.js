import React from "react";
import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import ChangeColorButton from "../ChangeColorButton";

describe("<ChangeColorButton />", () => {
	beforeEach(cleanup);
	afterEach(cleanup);

	it("should render without crashing", () => {
		render(<ChangeColorButton />);
	});

	test("button has the correct initial (red) color and initial ('Red') text", () => {
		render(<ChangeColorButton />);
		const button = screen.getByRole("button", { name: "Red" });
		expect(button).toHaveStyle({ backgroundColor: "#ff0000" });
	});

	test("button turns green color and 'Green' text after click", () => {
		render(<ChangeColorButton />);
		const button = screen.getByRole("button", { name: "Red" });
		fireEvent.click(button);

		expect(button).toHaveStyle({ backgroundColor: "#00ff00" });
		expect(button).toHaveTextContent("Green");
	});

	test("button returns to red after second click", () => {
		render(<ChangeColorButton />);
		const button = screen.getByRole("button", { name: "Red" });
		fireEvent.doubleClick(button);

		expect(button).toHaveStyle({ backgroundColor: "#ff0000" });
		// expect(button).toHaveTextContent("Red");
		expect(button.textContent).toBe("Red");
	});

	test("checkbox has the correct initial (red) color and initial ('Red') text", () => {
		render(<ChangeColorButton />);
		const checkbox = screen.getByRole("checkbox");
		expect(checkbox).not.toBeChecked();
	});

	test("checkbox turns green color and 'Green' text after click", () => {
		render(<ChangeColorButton />);
		const button = screen.getByRole("button", { name: "Red" });
		const checkbox = screen.getByRole("checkbox");

		fireEvent.click(checkbox);

		expect(checkbox).toBeChecked();
		expect(button).toHaveStyle({ backgroundColor: "#00ff00" });
		expect(button).toHaveTextContent("Green");
	});

	test("checkbox returns to red after second click", () => {
		render(<ChangeColorButton />);
		const button = screen.getByRole("button", { name: "Red" });
		const checkbox = screen.getByRole("checkbox");

		fireEvent.click(checkbox);
		fireEvent.click(checkbox);

		expect(checkbox).not.toBeChecked();
		expect(button).toHaveStyle({ backgroundColor: "#ff0000" });
		expect(button).toHaveTextContent("Red");
	});

	test("Matches snapshot", () => {
		const { asFragment } = render(<ChangeColorButton />);
		expect(asFragment()).toMatchSnapshot();
	});
});
