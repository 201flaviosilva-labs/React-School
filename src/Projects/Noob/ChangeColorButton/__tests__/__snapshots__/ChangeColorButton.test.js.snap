// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangeColorButton /> Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="NoobProject"
    title="Change Color Button"
  >
    <div
      class="NoobProjectHeader"
    >
      <h2>
        Change Color Button
      </h2>
    </div>
    <div
      class="NoobProjectMain NoobChangeColorButton"
    >
      <button
        style="background-color: rgb(255, 0, 0);"
      >
        Red
      </button>
      <label
        class="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
      >
        <span
          class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
         Red
      </label>
    </div>
  </div>
</DocumentFragment>
`;
